src/lib.rs

```diff
@@ -419,7 +420,7 @@ fn typecheck(thread: &Thread, filename: &Url, fileinput: &str) -> GluonResult<()
     let mut compiler = Compiler::new();
     // The parser may find parse errors but still produce an expression
     // For that case still typecheck the expression but return the parse error afterwards
-    let mut expr = match compiler.parse_partial_expr(&name, fileinput) {
+    let mut expr = match compiler.parse_partial_expr(&TypeCache::new(), &name, fileinput) {
```

Comment:
I wasn't sure if this was added to allow the language server to keep a cache somewhere to speed things up, so this might not be the desired implementation.

Reply:
I think I did the same in the other 2 PRs so it is fine. It may be named `TypeCache` but that may be a slightly optimistic name. It just holds some predefined `Arc`s to builtin types such as `Int`, `Bool`, `->` to ensure that at least these types will be shared, instead of one `Arc::new` per instance. I want to extend it to cache more things in the future though.

Reply:
Ah, that makes sense. Thanks for explaining!